File: dataset/human/python/file_2023.py
def main():
    N = int(input())
    S = input()
    mod = pow(10, 9) + 7
    d = dict()
    for c in S:
        d[c] = d.get(c, 0) + 1
    ans = 1
    for c in d:
        ans *= (d[c]+1)
        ans %= mod
    return (ans - 1) % mod


if __name__ == '__main__':
    print(main())

File: dataset/human/java/file_2023.java
import java.util.Scanner;

public class Main {
    public static void main(String args[]){
        Scanner sc=new Scanner(System.in);
        long n=sc.nextLong();
        long copyn = n;
        long ans = 0;
        for(int i=2;i<=Math.sqrt(n);i++) {
            int count=1;
            while(n % (Math.pow(i,count)) == 0) {
                n = (long) ( n / Math.pow(i,count));
                count++;
                ans++;
            }

            while(n % i == 0) {
                n = n / i;
            }
        }

        if(copyn == 1) {
            System.out.println(0);
        } else if(n>1) {
            System.out.println( 1 + ans);
        } else {
            System.out.println(ans);
        }

    }
}
